libs/ledger-live-common/src/onboarding/hooks/useOnboardingStatePolling.ts

```diff
@@ -80,7 +120,26 @@ export const useOnboardingStatePolling = ({
     setAllowedError,
     setFatalError,
     stopPolling,
+    getOnboardingStatePolling,
   ]);
 
   return { onboardingState, allowedError, fatalError };
 };
+
+const isDifferentOnboardingState = (
+  prevOnboardingState: OnboardingState | null,
+  newOnboardingState: OnboardingState | null
+): boolean => {
+  // OnboardingState are simple JSON-style objects, without methods
+  return (
+    JSON.stringify(prevOnboardingState) !== JSON.stringify(newOnboardingState)
+  );
+};
```

Comment:
Looks good. Another interesting thing i suggest is this same function can handle the entire logic and just return` onboardingStatePollingResult.onboardingState`

```
if(....){
return newOnboardingState
}

return prevOnboardingState
```